Plugins/BuildServerIntegration/AppVeyorIntegration/AppVeyorAdapter.cs

```diff
@@ -75,151 +73,93 @@ internal class AppVeyorAdapter : IBuildServerAdapter
 
             _buildServerWatcher = buildServerWatcher;
             _isCommitInRevisionGrid = isCommitInRevisionGrid;
-            var accountName = config.GetString("AppVeyorAccountName", null);
-            _accountToken = config.GetString("AppVeyorAccountToken", null);
-            var projectNamesSetting = config.GetString("AppVeyorProjectName", null);
-            if (string.IsNullOrWhiteSpace(accountName) && string.IsNullOrWhiteSpace(projectNamesSetting))
-            {
-                return;
-            }
-
+            string accountName = config.GetString("AppVeyorAccountName", null);
+            string accountToken = config.GetString("AppVeyorAccountToken", null);
             _shouldLoadTestResults = config.GetBool("AppVeyorLoadTestsResults", false);
 
             _fetchBuilds = new HashSet<ObjectId>();
 
-            _httpClientAppVeyor = GetHttpClient(WebSiteUrl, _accountToken);
+            _httpClientAppVeyor = GetHttpClient(WebSiteUrl, accountToken);
 
-            var useAllProjects = string.IsNullOrWhiteSpace(projectNamesSetting);
-            string[] projectNames = null;
-            if (!useAllProjects)
+            // projectId has format accountName/repoName
+            // accountName may be any accessible project (for instance upstream)
+            // if AppVeyorAccountName is set, projectNamesSetting may exclude the accountName part
+            string projectNamesSetting = config.GetString("AppVeyorProjectName", "");
+            var projectNames = _buildServerWatcher.ReplaceVariables(projectNamesSetting)
+                .Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries)
+                .Where(p => p.Contains("/") || !string.IsNullOrEmpty(accountName))
+                .Select(p => p.Contains("/") ? p : accountName.Combine("/", p))
+                .ToList();
+
+            if (projectNames.Count() == 0
+                && (string.IsNullOrWhiteSpace(accountName) || string.IsNullOrWhiteSpace(accountToken)))
             {
-                projectNames = _buildServerWatcher.ReplaceVariables(projectNamesSetting)
-                    .Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries);
+                // No projectIds in settings and no possibility to query
```

Comment:
```suggestion
                // No projectIds in settings, nothing to query
```

Reply:
// No projectIds in settings, cannot query